packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -252,14 +264,13 @@ export abstract class SharedObject<TEvent extends ISharedObjectEvents = ISharedO
         assert(!this._isSummarizing, 0x078 /* "Possible re-entrancy! Summary should not already be in progress." */);
         this._isSummarizing = true;
 
-        let gcData: IGarbageCollectionData;
         try {
             const serializer = new SummarySerializer(this.runtime.channelsRoutingContext);
             this.snapshotCore(serializer);
 
             // The GC data for this shared object contains a single GC node. The outbound routes of this node are the
             // routes of handles serialized during snapshot.
-            gcData = {
+            this._gcData = {
```

Comment:
this won't work with the sequence change. you should add getGCDataCore in this review as well

Reply:
Let me check-in that one and merge the changes here.

Reply:
+1 - needs to be addressed before merge.